keps/0023-enable-disable-features.md

```diff
@@ -0,0 +1,131 @@
+
+### Goals
+
+Given a feature, `A` is composed of Kubernetes object `x`, `y` and `z`
+ 
+- create or update kubernetes objects  `x`, `y` and `z`, when feature `A` is enabled.
+- delete kubernetes objects  `x`, `y` and `z`, when feature `A` is disabled.
+
+The goal of this KEP is to provide a way to create and delete objects `x`, `y` and `z`. 
+
+### Non-Goals
+
+- complex logic like loops to trigger multiple features
+- enable/disable nested features
+- evaluate multiple conditions to enable one feature
+
+## Proposal
+
+### Using Instance Parameters
+
+In KUDO, specific plans can be triggered when updating a parameter. Plans are specifically formed by phases, steps and utimately by tasks.
+And when talking about enabling/disabling features, we are eventually talking about the `Kind` of the Task. Currently, we have `Apply`, `Delete` and `Pipe` tasks kind.
+Enabling a feature is running a task of `Apply` kind and disabling is running a task of `Delete`
+
+We propose introducing a new Task kind named `Feature`. The Feature kind handles Apply or Delete resources based on the value of a parameter in the Operator Instance.
+
+A `Feature` task will look like:
+
+```
+tasks:
+  - name: mirror-maker-configuration
+    kind: Feature
+    spec:
+      parameter: MIRROR_MAKER_ENABLED
+      resources:
+        - mirror-maker-cm.yaml
+  - name: mirror-maker
+    kind: Feature
+    spec:
+      parameter: MIRROR_MAKER_ENABLED
+      resources:
+        - mirror-maker.yaml
+```
+
+
+the parameter used in task would be defined in the parameters for the Operator, and must evaluate to a boolean value.
+```
+apiVersion: kudo.dev/v1beta1
+parameters:
+  - name: MIRROR_MAKER_ENABLED
+    default: "false"
+    trigger: mirror-maker-plan
+```
+
+And the plan triggered by the parameter will be the one using `Feature` tasks in their steps.
+
+```
+plans:
+  mirror-maker-plan:
+    strategy: serial
+    phases:
+      - name: deploy-mirror-maker
+        strategy: serial
+        steps:
+          - name: configuration
+            tasks:
+              - mirror-maker-configuration
+          - name: deploy
+            tasks:
+              - mirror-maker
+```
+
+In the tasks is where we can use the `Feature` kind. Which will apply the resources rendered by templates `mirror-maker.yaml` and `mirror-maker-cm.yaml` in case the value of parameter `MIRROR_MAKER_ENABLED` is true. 
+In case, the value is `false` it will delete the resources rendered by `mirror-maker.yaml` and `mirror-maker-cm.yaml`. 
+
+In case the parameter value is **true**:
+
+If there are objects to created the Task will be marked successful, once those are created.
+If there are no objects to be created the Task will be marked as successful. 
+
+In case the parameter value is **false**:
+
+If there are objects to be deleted the Task will be marked as successful, once those are marked for Deletion. 
```

Comment:
so it basically does client.Delete for all resources specified in the resources of that task? Maybe that's a better description? WDYT?